gitpod-io/gitpod
# Add `x-client` in ide-metrics component

## Description

Add `x-client` a global setting in ide-metrics

## TODOs
- [ ] ‼️ After this PR, supervisor public api metrics will be broken, so we need to change ide supervisor public api SLO

## Related Issue(s)

Fixes #

## How to test


Check also test results https://github.com/gitpod-io/gitpod/pull/16701#issuecomment-1457867144

- Open workspace in preview env https://hw-metrics-client.preview.gitpod-dev.com/workspaces
- `ide-metrics` in preview env should correctly setup `metric_client` for supervisor public api metrics

**debug mode**
- Open this PR with Gitpod
- Exec commands below
```
cd components/ide-metrics
# start metric server
go run main.go run --config config-example.json --json-log=false --verbose

# post curl
curl -X "POST" "http://localhost:3000/metrics-api/metrics/counter/add/gitpod_test_counter" \
     -H 'x-client: vscode' \
     -H 'Content-Type: application/json; charset=utf-8' 
```
- `/metrics` endpoint at port 9500 should have `metric_client` field with vscode

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
